src/Cryptol/TypeCheck/Sanity.hs

```diff
@@ -71,7 +71,7 @@ checkType ty =
     TVar tv -> lookupTVar tv
 
     TRec fs ->
-      do forM_ fs $ \(_,t) ->
+      do forM_ (canonicalFields fs) $ \(_,t) ->
```

Comment:
Is `canonicalFields` even necessary here? Doesn't `forM_` work for any `Traversable` type, including `RecordMap`?